samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,514 @@
+
+    private val _assetGroupNameState = MutableStateFlow("<Nothing selected>")
+    var assetGroupNameState: StateFlow<String> = _assetGroupNameState
+
+    private val _assetTypeNameState = MutableStateFlow("<Nothing selected>")
+    val assetTypeNameState: StateFlow<String> = _assetTypeNameState
+
+    // Represents the snap source settings object (for enabling and disabling), along with a name
+    // to use in the UI, and a symbol swatch
+    data class SnapSourceProperty(
+        val name: String,
+        val swatch: BitmapDrawable,
+        val snapSourceSettings: SnapSourceSettings
+    )
+    private val _snapSourcePropertyList = MutableStateFlow(listOf<SnapSourceProperty>())
+    internal val snapSourcePropertyList: StateFlow<List<SnapSourceProperty>> =
+        _snapSourcePropertyList
+
+    // Create boolean flags to track the state of UI components
+    private val _isEditButtonEnabled = MutableStateFlow(false)
+    internal val isEditButtonEnabled = _isEditButtonEnabled.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            arcGISMap.load().onFailure { error ->
+                messageDialogVM.showMessageDialog(
+                    "Failed to load map",
+                    error.message.toString()
+                )
+            }
+
+            // Load the mobile map package
+            geodatabase.load().onSuccess {
+                // Add layers from the geodatabase to the map
+                addLayersToMapFromGeodatabase(application)
+
+                // Set the utility network on the map and load it
+                arcGISMap.utilityNetworks.add(geodatabase.utilityNetworks.first())
+                arcGISMap.utilityNetworks.first().load().onFailure {
+                    messageDialogVM.showMessageDialog(
+                        "Error loading utility network", it.message.toString()
+                    )
+                }
+
+                // Set up symbol swatches
+                symbolSwatches[SnapRuleBehavior.RulesPreventSnapping] = createSwatch(rulesPreventSymbol)
+                symbolSwatches[SnapRuleBehavior.RulesLimitSnapping] = createSwatch(rulesLimitSymbol)
+                symbolSwatches[SnapRuleBehavior.None] = createSwatch(noneSymbol)
+            }.onFailure {
+                messageDialogVM.showMessageDialog(
+                    "Error loading geodatabase", it.message.toString()
+                )
+            }
+        }
+    }
+
+    /**
+     * Start the geometry editor to edit the geometry of the selected feature.
+     */
+    fun editFeatureGeometry() {
+        // Get the symbol for the selected feature
+        selectedFeature?.let { feature ->
+
+            // Use the symbol from the selected feature in the style of the geometry editor tool
+            val symbol =
+                (feature.featureTable as GeodatabaseFeatureTable).layerInfo?.drawingInfo?.renderer?.getSymbol(
+                    feature
+                )
```

Comment:
Seems like `featureTable` will be needed in the code below, we could keep track of the value and simplify the `symbol` code a bit. Also, we need to use `as?` for all unchecked null-casts to ensure null-safety:  
```suggestion
            // Get the geodatabase feature table of the selected feature
            val featureTable = (feature.featureTable as? GeodatabaseFeatureTable) ?: return
            // Use the symbol from the selected feature in the style of the geometry editor tool
            val symbol = featureTable.layerInfo?.drawingInfo?.renderer?.getSymbol(feature)
```